File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = nums[0];
        int sum = nums[0];

        for(int i=1; i<nums.length; i++){
            sum = Math.max(nums[i], sum+nums[i]);
            maxSum = Math.max(sum, maxSum);
        }
        return maxSum;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curr = nums[0]
        maxi = nums[0]

        for i in range(1, len(nums)):
            curr = max(nums[i], nums[i]+curr)
            maxi = max(maxi, curr)

        return maxi